data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/json/serializers/network_action/InjectionSetPointSerializer.java

```diff
@@ -6,24 +6,33 @@
  */
 package com.farao_community.farao.data.crac_impl.json.serializers.network_action;
 
-import com.farao_community.farao.data.crac_api.ExtensionsHandler;
-import com.farao_community.farao.data.crac_impl.remedial_action.network_action.InjectionSetpoint;
+import com.farao_community.farao.data.crac_impl.remedial_action.network_action.InjectionSetpointImpl;
 import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.type.WritableTypeId;
+import com.fasterxml.jackson.databind.JsonSerializer;
 import com.fasterxml.jackson.databind.SerializerProvider;
-import com.powsybl.commons.json.JsonUtil;
+import com.fasterxml.jackson.databind.jsontype.TypeSerializer;
 
 import java.io.IOException;
 
-import static com.farao_community.farao.data.crac_impl.json.JsonSerializationNames.SETPOINT;
+import static com.farao_community.farao.data.crac_impl.json.JsonSerializationNames.*;
 
 /**
  * @author Sebastien Murgey {@literal <sebastien.murgey at rte-france.com>}
  */
-public class InjectionSetPointSerializer extends NetworkActionSerializer<InjectionSetpoint> {
+public class InjectionSetPointSerializer extends JsonSerializer<InjectionSetpointImpl> {
```

Comment:
InjectionSetPointImplSerializer ? I'm actually not sure that Impl is relevant for the ElementaryActions

Reply:
I added the interfaces in Crac API to make it more lisible

Reply:
We'll fix the naming later when moving serializers to the API